atom/browser/ui/cocoa/views_delegate_mac.mm

```diff
@@ -16,7 +16,18 @@
 void ViewsDelegateMac::OnBeforeWidgetInit(
     views::Widget::InitParams* params,
     views::internal::NativeWidgetDelegate* delegate) {
-  DCHECK(params->native_widget);
+  // If we already have a native_widget, we don't have to try to come
+  // up with one.
+  if (params->native_widget)
+    return;
+
+  if (!native_widget_factory().is_null()) {
+    params->native_widget = native_widget_factory().Run(*params, delegate);
+    if (params->native_widget)
+      return;
+  }
+
+  params->native_widget = nullptr;
```

Comment:
I don't understand this, why would it have been `DCHECK`ed to be present before but it's OK for it to be `nullptr` now?